### ISSUE TITLE
Firefox: Assertion `0 < t->stop_sig()' failed to hold. Replaying `SCHED': expecting tracee signal or trap, but instead at `<unknown-syscall--1>'

### ISSUE THREAD

**Issue by hotsphink** [mozilla/rr#2004]
I have a recording of a run of ./mach mochitest addon-sdk/source/test/test-disposable.js, with calls to Math.sin() and Math.cos() added so that I can set breakpoints at convenient points. I set a breakpoint on js::math_sin, then continue. It bombs out with

[FATAL /home/sfink/src/rr/src/ReplaySession.cc:468:check_pending_sig() errno: SUCCESS] 
 (task 7894 (rec:28442) at time 440255)
 -> Assertion `0 < t->stop_sig()' failed to hold. Replaying `SCHED': expecting tracee signal or trap, but instead at `<unknown-syscall--1>' (ticks: 403060630)

[rr-session.txt](https://github.com/mozilla/rr/files/859864/rr-session.txt)

Output of |rr dump -b -m -p| around global_time 440255:

[rr-dump-portion.txt.gz](https://github.com/mozilla/rr/files/859869/rr-dump-portion.txt.gz)

dbaron reported the same assertion on the #rr channel, saying it was reproducible with a given recording but did not happen with all recordings.

**Comment by hotsphink** [comment]
This is with rr 4.5.0 at ee20c58af947ad05b67bc27966d96248be74e365

**Comment by rocallahan** [comment]
Does this reproduce on every recording of that test?

**Comment by hotsphink** [comment]
I don't think so? I know it has happened on more than one recording. But I don't think it happened on all of them.

Rerunning on a different recording right now to check.

**Comment by hotsphink** [comment]
I know I have been able to make it quite a ways into that problematic recording without hitting it. (Enough to track down my problem in 2 days, so I was happy.) I don't know if it depends on following the same sequence of actions or not to hit it. I was starting up multiple instances and varying my actions slightly in order to try to make progress.

**Comment by hotsphink** [comment]
Hm. It looks like I don't have any more runs of that exact code. But I do have other runs of the identical compiled code, just with a different test js file. I tried reproducing there (harder because I don't have Math.sin or Math.cos present in this one), but soon hit this first:

(rr) info local
No locals.
(rr) n
../../gdb/infrun.c:7193: internal-error: int switch_back_to_stepped_thread(execution_control_state*): Assertion `!schedlock_applies (tp)' failed.
A problem internal to GDB has been detected,
further debugging may prove unreliable.
Quit this debugging session? (y or n) 

It didn't seem all that fatal if I continued, but I haven't been able to get that original assertion in this session.

**Comment by hotsphink** [comment]
Hm. It looks like I don't have any more runs of that exact code. But I do have other runs of the identical compiled code, just with a different test js file. I tried reproducing there (harder because I don't have Math.sin or Math.cos present in this one), but soon hit this first:

(rr) info local
No locals.
(rr) n
../../gdb/infrun.c:7193: internal-error: int switch_back_to_stepped_thread(execution_control_state*): Assertion `!schedlock_applies (tp)' failed.
A problem internal to GDB has been detected,
further debugging may prove unreliable.
Quit this debugging session? (y or n) 

It didn't seem all that fatal if I continued, but I haven't been able to get that original assertion in this session.
